userCode/main.py

```diff
@@ -585,13 +586,32 @@ def nquads_to_renci(
 
 
 @schedule(
-    cron_schedule="@daily",
+    cron_schedule="@weekly",
     job=harvest_job,
-    default_status=DefaultScheduleStatus.STOPPED,
+    default_status=DefaultScheduleStatus.STOPPED
+    if RUNNING_AS_TEST_OR_DEV
+    else DefaultScheduleStatus.RUNNING,
 )
-def crawl_entire_graph_schedule():
-    for partition_key in sources_partitions_def.get_partition_keys():
-        yield RunRequest(partition_key=partition_key)
+def crawl_entire_graph_schedule(context):
+    get_dagster_logger().info("Schedule triggered.")
+
+    result = materialize([gleaner_config], instance=context.instance)
+    if not result.success:
+        raise Exception(f"Failed to materialize gleaner_config!: {result}")
+
+    partition_keys = context.instance.get_dynamic_partitions("sources_partitions_def")
+    get_dagster_logger().info(f"Partition keys: {partition_keys}")
+
+    if partition_keys:
+        for partition_key in partition_keys:
+            yield RunRequest(
+                job_name="harvest_source",
+                run_key="havest_weekly",
+                partition_key=partition_key,
+                tags={"run_type": "harvest_weekly"},
+            )
+    else:
+        get_dagster_logger().warning("No partition keys found.")
```

Comment:
Think this should be an exception; no partitions means something went wrong
```suggestion
    else:
        raise RuntimeError("No partition keys found.")
```